### PR TITLE
Further cleanup addressing more copied code

### PR DESCRIPTION
**Pull request by daniel-beck**
Extract an interface with default method implementations common to the strategy and the properties (as opposed to the existing one that only did code common to properties), an interface for their descriptors, and a new abstract converter.

Untested.

### CODE REVIEW

**Comment by bpedersen2** [comment]
Fixes JENKINS-46923

File: src/main/java/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty.java [hunk 1]
```diff
@@ -96,6 +96,7 @@ public AuthorizationMatrixProperty(Map<Permission,? extends Set<String>> granted
             this.grantedPermissions.put(e.getKey(),new HashSet<>(e.getValue()));
     }
 
+    @Restricted(NoExternalUse.class)
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty.java]: Why? I understand that nobody is expected to write to this method. `Collections.unmodifyableSet()` would be reasonable, but I do not see issues with having getter in API

Comment (daniel-beck) [reply src/main/java/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty.java]: This only exists to build up the potential groups for http://javadoc.jenkins.io/hudson/security/AuthorizationStrategy.html#getGroups-- and is not intended for any other use.

Comment (oleg-nenashev) [reply src/main/java/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty.java]: Collections.unmodifyableSet() would be useful anyway

Comment (daniel-beck) [reply src/main/java/com/cloudbees/hudson/plugins/folder/properties/AuthorizationMatrixProperty.java]: I went with a copy of the set instead, consistent with some other getters that return a newly built collection.

**Comment by daniel-beck** [comment]
Merging to get a beta-2 out, further feedback is always appreciated @oleg-nenashev

**Pull request merged**